src/ArduinoIoTCloudLPWAN.h

```diff
@@ -0,0 +1,87 @@
+/*
+   This file is part of ArduinoIoTCloud.
+
+   Copyright 2019 ARDUINO SA (http://www.arduino.cc/)
+
+   This software is released under the GNU General Public License version 3,
+   which covers the main part of arduino-cli.
+   The terms of this license can be found at:
+   https://www.gnu.org/licenses/gpl-3.0.en.html
+
+   You can be released from the requirements of the above licenses by purchasing
+   a commercial license. Buying such a license is mandatory if you want to modify or
+   otherwise use the software for commercial activities involving the Arduino
+   software without disclosing the source code of your own applications. To purchase
+   a commercial license, send an email to license@arduino.cc.
+*/
+
+#ifndef ARDUINO_IOT_CLOUD_LPWAN_H
+#define ARDUINO_IOT_CLOUD_LPWAN_H
+
+#include <ArduinoIoTCloud.h>
+#include <Arduino_ConnectionHandler.h>
+
+static uint8_t const DEFAULT_CBOR_LORA_MSG_SIZE = 255;
+
+class ArduinoIoTCloudLPWAN : public ArduinoIoTCloudClass {
+  public:
+    ArduinoIoTCloudLPWAN();
+    ~ArduinoIoTCloudLPWAN();
+    int connect();
+    bool disconnect();
+    int connected();
+    inline void update() {
```

Comment:
`ArduinoIoTCloud` should be an interface for future implementation of other protocols, in this case they are almost the same but tomorrow could not

Reply:
I have to disagree here because
`inline void update(int const reconnectionMaxRetries, int const reconnectionTimeoutMs) __attribute__((deprecated))`
and
`void update(CallbackFunc onSyncCompleteCallback) __attribute__((deprecated))`
will be removed soon since they have been marked `__attribute__((deprecated))` for some time. What remains would be a pure virtual `update` in `ArduinoIoTCloud` which is then implemented in the derived base classes. However, let's not do this change now.